client/app/dbaas/logs/detail/streams/home/logs-streams-home.controller.js

```diff
@@ -40,6 +49,35 @@ class LogsStreamsHomeCtrl {
             streamId: stream.streamId
         });
     }
+
+    /**
+     * show delete stream confirmation modal
+     *
+     * @param {any} stream to delete
+     * @memberof LogsStreamsHomeCtrl
+     */
+    showDeleteConfirm (stream) {
+        this.CloudMessage.flushChildMessage();
+        return this.ControllerHelper.modal.showDeleteModal({
+            titleText: this.$translate.instant("logs_stream_delete_title"),
+            text: this.$translate.instant("logs_stream_delete_message", { stream: stream.title })
+        }).then(this.delete(stream));
```

Comment:
() => this.delete(stream)... I'm surprised .then(this.delete(stream)) is working :/

Reply:
Yes it works. I have tested it. But there is a strange problem, some times even then will be executed when user selects cancel button :(. What do you use this in other pages?

Reply:
i replaced it with .then(() => this.delete(stream))
by mistake i put  .then(this.delete(stream))